lib/spack/spack/ci.py

```diff
@@ -1184,6 +1196,44 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
 
             output_object['cleanup'] = cleanup_job
 
+        if 'signing-job-attributes' in gitlab_ci:
+            # External signing: generate a job to check and sign binary pkgs
+            stage_names.append('stage-sign-pkgs')
+            signing_job_config = gitlab_ci['signing-job-attributes']
+            signing_job = {}
+
+            signing_job_attrs_to_copy = [
+                'image',
+                'tags',
+                'variables',
+                'before_script',
+                'script',
+                'after_script',
+            ]
+
+            _copy_attributes(signing_job_attrs_to_copy,
+                             signing_job_config,
+                             signing_job)
+
+            signing_job_tags = []
+            if 'tags' in signing_job:
+                signing_job_tags = [tag for tag in signing_job['tags']
+                                    if tag not in spack_reserved_tags]
```

Comment:
I think we perform this filtering enough times that we could benefit from extracting it into a function